src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -450,13 +429,11 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+                var binder = compilation.GetBinder((CSharpSyntaxNode)_containingType.DeclaringSyntaxReferences[0].SyntaxTree.GetRoot());
```

Comment:
@TyOverby 

I'm still a bit confused on why we are getting the syntax tree in this manner.  It appears that the `ScriptEntryPoint` type is only instantiated in one place in our code base:

https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs#L50

One of the parameters to this function is an instance of `SynthesizedInteractiveInitializerMethod` which contains a `SourceMemeberContainerSymbol`.  Can't we just grab the syntax tree off of that?  At a glance it seems more appropriate than grabbing an arbitrary tree here.